lightning/src/chain/keysinterface.rs

```diff
@@ -36,9 +38,61 @@
 use ln::chan_utils::{HTLCOutputInCommitment, make_funding_redeemscript, ChannelPublicKeys, HolderCommitmentTransaction, ChannelTransactionParameters, CommitmentTransaction};
 use ln::msgs::UnsignedChannelAnnouncement;
 
+use std::collections::HashSet;
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::io::Error;
-use ln::msgs::DecodeError;
+use ln::msgs::{DecodeError, MAX_VALUE_MSAT};
+
+/// Information about a spendable output to a P2WSH script. See
+/// SpendableOutputDescriptor::DynamicOutputP2WSH for more details on how to spend this.
+#[derive(Clone, Debug, PartialEq)]
+pub struct DynamicP2WSHOutputDescriptor {
+	/// The outpoint which is spendable
+	pub outpoint: OutPoint,
+	/// Per commitment point to derive delayed_payment_key by key holder
+	pub per_commitment_point: PublicKey,
+	/// The nSequence value which must be set in the spending input to satisfy the OP_CSV in
+	/// the witness_script.
+	pub to_self_delay: u16,
+	/// The output which is referenced by the given outpoint
+	pub output: TxOut,
+	/// The revocation_pubkey used to derive witnessScript
+	pub revocation_pubkey: PublicKey,
+	/// Arbitrary identification information returned by a call to
+	/// `ChannelKeys::key_derivation_params()`. This may be useful in re-deriving keys used in
+	/// the channel to spend the output.
+	pub key_derivation_params: (u64, u64),
```

Comment:
it seems useful to also provide funding outpoint/channel ID, so that `KeysManager` implementations can match these to channels in a more intuitive way

Reply:
Note this is really in #786. Should we instead have 4 u64s or maybe `[u8; 32]`? I'm a bit hesitant to make this anything other than arbitrary data, since its totally up to the client what they want to use it for.

Reply:
Sorry about the misplaced comments.

In any case, `[u8; 32]` seems perfect and communicates that it's an opaque ID.  Maybe also change the name to `key_derivation_id`?

Reply:
I went with `channel_keys_id`. See #786.